lib/ES/Repo.pm

```diff
@@ -225,8 +226,10 @@ sub _prepare_sub_dir {
         run qw(git remote add subbed_repo), $source_root;
         run qw(git fetch subbed_repo), $subbed_head;
         run qw(git remote remove subbed_repo);
-        run qw(git config core.sparseCheckout true);
-        $self->_write_sparse_config( $work, $path );
+        unless ($path eq '.') {
+            run qw(git config core.sparseCheckout true);
```

Comment:
so many small git-fixes 😄

Reply:
> so many small git-fixes smile

I knew this'd be difficult to get right! I'm so glad I can write all these tests for it.